data/src/main/java/org/oppia/data/backends/gae/ModifyJsonInterceptor.kt

```diff
@@ -0,0 +1,30 @@
+package org.oppia.data.backends.gae
+
+import okhttp3.*
+
+import java.io.IOException
+
+class ModifyJsonInterceptor : Interceptor {
+
+  @Throws(IOException::class)
+  override fun intercept(chain: Interceptor.Chain): Response {
+    val request = chain.request()
+    val response = chain.proceed(request)
+
+    if (response.code() == 200) {
+      if (response.body() != null) {
+        var rawJson = response.body()!!.string()
+        if (rawJson.startsWith(NetworkSettings.XSSI_PREFIX)) {
+          rawJson = rawJson.substring(rawJson.indexOf('\n') + 1)
```

Comment:
Should we maybe just remove the XSSI_PREFIX instead? Extra whitespace should just be ignored by the JSON parser, and this guards against the assumption that a newline needs to be present for the prefix.

Reply:
Code updated and used removePrefix method.